daemon/volumes.go

```diff
@@ -205,7 +205,11 @@ func (daemon *Daemon) registerMountPoints(container *container.Container, hostCo
 			if cv, ok := v.(interface {
 				CachedPath() string
 			}); ok {
-				mp.Source = cv.CachedPath()
+				if cfg.VolumeOptions != nil && cfg.VolumeOptions.Subpath != "" {
```

Comment:
I'm extreemly interested in this (actually mounting named volumes with sub-directories) but I have no clue how to use the moby project. Can someone sot some light into this?